### PR TITLE
More suggestions to `postTransactionFee`

### PR DESCRIPTION
**Pull request by Anviking**
- [ ] Base branch needs to be rebased

### CODE REVIEW

File: lib/wallet/src/Cardano/Wallet.hs [hunk 1]
```diff
@@ -2928,7 +2928,7 @@ transactionFee
     -> TransactionCtx
     -> PreSelection
     -> ExceptT ErrSelectAssets IO (Percentile 10 Fee, Percentile 90 Fee)
-transactionFee DBLayer{atomically, walletsDB} protocolParams
+transactionFee DBLayer{atomically, walletsDB} protocolParams'
```
Comment (Unisay) [inline lib/wallet/src/Cardano/Wallet.hs]:
```suggestion
transactionFee DBLayer{atomically, walletsDB} protocolParams
```

File: lib/wallet/src/Cardano/Wallet.hs [hunk 2]
```diff
@@ -2939,12 +2939,13 @@ transactionFee DBLayer{atomically, walletsDB} protocolParams
                 Just ws -> pure $ WalletState.getLatest ws
         let utxoIndex = UTxOIndex.fromMap . CS.toInternalUTxOMap $
                 availableUTxO @s mempty wallet
+        let protocolParams = toBalanceTxPParams protocolParams'
```
Comment (Unisay) [inline lib/wallet/src/Cardano/Wallet.hs]:
```suggestion
        let protocolParamsForBalancing = toBalanceTxPParams protocolParams
```

File: lib/wallet/src/Cardano/Wallet.hs [hunk 3]
```diff
@@ -2939,12 +2939,13 @@ transactionFee DBLayer{atomically, walletsDB} protocolParams
...
         pureTimeInterpreter <- lift $ snapshot ti
         unsignedTxBody <- liftIO $
             either (throwIO . ExceptionConstructTx . ErrConstructTxBody) pure $
                 mkUnsignedTransaction txLayer @era
                     (unsafeShelleyOnlyGetRewardXPub @s @k @n (getState wallet))
-                    protocolParams
+                    protocolParams'
```
Comment (Unisay) [inline lib/wallet/src/Cardano/Wallet.hs]:
```suggestion
                    protocolParamsForBalancing
```

**Comment by Anviking** [comment]
Merged manually

**Pull request closed without merging**